src/core/simulation.jl

```diff
@@ -225,20 +231,57 @@ function _check_steps(
     stage_initial_times::Dict{Int, Vector{Dates.DateTime}},
 )
     for (stage_number, stage_name) in sim.sequence.order
-        forecast_count = length(stage_initial_times[stage_number])
         stage = sim.stages[stage_name]
-        if get_steps(sim) * get_executions(stage) > forecast_count
+        execution_counts = get_executions(stage)
+        @assert length(findall(x -> x == stage_number, sim.sequence.execution_order)) ==
+                execution_counts
+        forecast_count = length(stage_initial_times[stage_number])
+        if get_steps(sim) * execution_counts > forecast_count
             throw(IS.ConflictingInputsError("The number of available time series ($(forecast_count)) is not enough to perform the
             desired amount of simulation steps ($(sim.steps*stage.internal.execution_count))."))
         end
     end
     return
 end
 
+function _check_required_ini_cond_caches(sim::Simulation, ::IntraStageChronology)
+    for (stage_number, stage_name) in sim.sequence.order
+        stage = get_stage(sim, stage_name)
+        for (k, v) in get_initial_conditions(stage.internal.psi_container)
+            isnothing(v[1].cache_type) && continue
+            c = get_cache(stage, v[1].cache_type)
+            if isnothing(c)
+                throw(IS.ArgumentError("Cache $(v[1].cache_type) not defined for initial condition $(k.ic_type) in stage $stage_name"))
+            end
+        end
+    end
+    return
+end
+
+function _check_required_ini_cond_caches(sim::Simulation, ::InterStageChronology)
+    for (stage_number, stage_name) in sim.sequence.order
+        receiving_stage = get_stage(sim, stage_name)
+        for (k, v) in get_initial_conditions(receiving_stage.internal.psi_container)
+            # No cache needed for the initial condition -> continue
+            isnothing(v[1].cache_type) && continue
+            c = nothing
+            # Search other stages
+            for source_stage in values(sim.stages)
+                c = get_cache(source_stage, v[1].cache_type)
+                break
```

Comment:
Looping one time and then breaking feels odd.  And it doesn't match the comment.  If this is what you want then you could write `source_stage = iterate(values(sim.stages))[1]`.  Then you could delete `c = nothing`.

Reply:
if c keeps being nothing because the `get_cache` functions return nothing it means that it is not defined. I don't think the proposed change achieves that.

Reply:
fixed in 2c30f4a.